lib/uxtx.js

```diff
@@ -1,664 +1,715 @@
-            // if controlled by wallet
-            if (o.path)
-              return !o.path.change;
-              // handle sending to controlled wallet
-
-            return true; // all remaining outputs included
-          });
-        const recipients = outputs.map(o => o.address);
-
-        // parse counterpary based on number of recipients
-        if (recipients.length === 1)
-          counterparty = recipients[0];
-        else if (recipients.length > 1)
-          counterparty = labels.MULTIPLE_ADDRESS;
-        else
-          counterparty = labels.UNKNOWN_ADDRESS;
-
-        break;
-      case this.TYPES.COINBASE:
-        // assume one input and one output
-        counterparty = json.outputs[0].address;
-
-        break;
-      case this.TYPES.UNKNOWN:
-      default:
-        counterparty = labels.UNKNOWN_ADDRESS;
-    }
-
-    this._counterparty = counterparty;
-    return counterparty;
-  }
-
-  /*
-   * Get tx recipients
-   * @returns {String}
-   */
-  getRecipients() {
-    if (this._recipients)
-      return this._recipients;
-
-    let recipients;
-
-    const labels = this.getLabels();
-    const json = this.getJSON();
-
-    switch(this.getUXType()) {
-      case this.TYPES.DEPOSIT: // receive transaction
-        // parse recipients
-        // TODO: catch edge cases here around more complex txns
-        recipients = json.inputs.map(i => i.address)
-
-        break;
-      case this.TYPES.WITHDRAW: // send transaction
-        // parse recipients
-        const outputs= json.outputs
-          .filter(o => {
-            // only will have path property
-            // if controlled by wallet
-            if (o.path)
-              return !o.path.change;
-              // handle sending to controlled wallet
-
-            return true; // all remaining outputs included
-          })
-          .map(o => o.address);
-
-        break;
-      case this.TYPES.COINBASE:
-        // assume one input and one output
-        recipients = [json.outputs[0].address];
-
-        break;
-      case this.TYPES.UNKNOWN:
-      default:
-        recipients = [];
-    }
-
-    this._recipients = recipients;
-    return recipients;
-  }
-
-  /*
-   * Get human readable labels
-   * These can be configured with this.fromOptions
-   * @param {String} label - return a specific labels value
-   * @returns {Object|String}
-   */
-  getLabels(label) {
-    if (label)
-      return this._labels[label];
-    return this._labels;
-  }
-
-  /*
-   * Get human readable labels
-   * These can be configured with this.fromOptions
-   * @static
-   * @param {String|Buffer} data - raw tx data
-   * @param {String} enc - encoding raw tx data is in
-   * @param {Objects} options - UXTX options
-   * @returns {Object|String}
-   */
   static fromRaw(data, enc, options) {
     if (typeof data === 'string')
       data = Buffer.from(data, enc);
-    return new this(options).fromRaw(data);
+    return new this().fromRaw(data, enc, options);
```

Comment:
I'm still playing around with this but I think this is causing problems as implemented with bwallet and handshake. In [txManager](https://github.com/bpanel-org/bpanel-utils/blob/e02f76886525bbec0f4a5bf6ee2d7ce2068f6047/lib/txManager.js#L78) it uses the static method on the class, but this won't account for the SuperClass. Not sure which is the best approach around this. Either we don't support the static `fromRaw` since it's more important to instantiate with the chain and create the SuperClass? And then change the use in txManager maybe? Will test further to confirm but wanted to put this here to start the discussion.

Reply:
nvm, I got it to work w/ bwallet and handshake by passing `options` to `this()` 👍

Reply:
this is the only change required for you to get it to work?